### PR TITLE
[DPSTAT-837] Upgrade micronaut to verison 4.3.7

### PR DESCRIPTION
**Pull request by skykanin**
Upgrade micronaut to major version 4 and apply recommended changes from their migration guide:
https://micronaut.io/2023/05/09/upgrade-to-micronaut-framework-4-0-0

PS: `dapla-dpl-psuedo-core` has also been upgraded to micronaut v4. Relevant PR found [here](https://github.com/statisticsnorway/dapla-dlp-pseudo-core/pull/37)

### CODE REVIEW

File: src/main/java/no/ssb/dlp/pseudo/service/accessgroups/CloudIdentityService.java [hunk 1]
```diff
@@ -2,16 +2,19 @@
...
+import jakarta.inject.Singleton;
 import lombok.RequiredArgsConstructor;
 
-import javax.inject.Singleton;
 import java.util.ArrayList;
 import java.util.List;
 
 @Singleton
-@RequiredArgsConstructor
+// @RequiredArgsConstructor
```
Comment (kschulst) [inline src/main/java/no/ssb/dlp/pseudo/service/accessgroups/CloudIdentityService.java:12]: Enable this and remove constructor?

File: src/main/java/no/ssb/dlp/pseudo/service/pseudo/PseudoSecrets.java [hunk 2]
```diff
@@ -21,9 +21,13 @@ public class PseudoSecrets {
...
      * @param configuredPseudoSecrets pseudo secrets coming from the config or environment
      */
-    public PseudoSecrets(SecretService secretService, @Property(name = "pseudo.secrets") Map<String, PseudoSecret> configuredPseudoSecrets) {
+    public PseudoSecrets(
+            SecretService secretService,
+            @Property(name = "pseudo.secrets")
+            Map<String, PseudoSecret> configuredPseudoSecrets) {
         this.secretService = secretService;
         this.configuredPseudoSecrets = Optional.ofNullable(configuredPseudoSecrets).orElse(Map.of());
+        System.out.println("TESTING");
```
Comment (kschulst) [inline src/main/java/no/ssb/dlp/pseudo/service/pseudo/PseudoSecrets.java:30]: Remove this?

File: src/main/java/no/ssb/dlp/pseudo/service/pseudo/PseudoServiceTypeConverterRegistrar.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+import no.ssb.dlp.pseudo.core.typeconverter.PseudoSecretTypeConverter;
+
+import java.util.Map;
+
+/**
+ *  Micronaut {@link io.micronaut.core.convert.TypeConverter} implementations defined in a different
+ *  package like the ones from {@link no.ssb.dlp.pseudo.core} need to be manually registered
+ *  using this {@link io.micronaut.core.convert.TypeConverterRegistrar} implementation.
+ *
+ * @author Nicholas Jaunsen
```
Comment (kschulst) [inline src/main/java/no/ssb/dlp/pseudo/service/pseudo/PseudoServiceTypeConverterRegistrar.java:18]: We should not use @author tags as they promote/signals personal ownership in code. Information is already available in git.

We should probably have an ADR for this.

File: src/main/java/no/ssb/dlp/pseudo/service/sid/ExternalSidService.java [hunk 4]
```diff
@@ -5,18 +5,22 @@
...
 
-import javax.inject.Singleton;
+import jakarta.inject.Singleton;
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
 
 @Singleton
-@RequiredArgsConstructor
+//@RequiredArgsConstructor
```
Comment (kschulst) [inline src/main/java/no/ssb/dlp/pseudo/service/sid/ExternalSidService.java:14]: Enable this annotation and let Lombok generate constructors instead

**Review by kschulst — APPROVED** [review]
Nice work. Looks good, and I like that we are introducing records.

I added a couple of minor comments that you might want to fix before merging, but green lights from me 👍

File: src/main/java/no/ssb/dlp/pseudo/service/pseudo/PseudoServiceTypeConverterRegistrar.java (hunk 3 above)
Comment (skykanin) [reply src/main/java/no/ssb/dlp/pseudo/service/pseudo/PseudoServiceTypeConverterRegistrar.java]: What's an ADR?

**Review by bjornandre — DISMISSED** [review]

**Review by Andilun — APPROVED** [review]

**Pull request merged**